controller/controller.go

```diff
@@ -275,6 +288,19 @@ func (c *controller) updateIngresses() error {
 	return nil
 }
 
+func (c *controller) validIngressName(ingress *v1beta1.Ingress) bool {
+
+	isValid := false
+
+	if ingressName, ok := ingress.Annotations[ingressNameAnnotation]; ok {
+		isValid = ingressName == c.ingressName
+	} else if c.mainIngress {
+		isValid = true
+	}
+
+	return isValid
+}
```

Comment:
I think this could be simplified as
```suggestion
        if ingressName, ok := ingress.Annotations[ingressNameAnnotation]; ok {
		return ingressName == c.ingressName
	}
        return c.mainIngress
```